src/components/NewHathorAlert.js

```diff
@@ -34,41 +35,80 @@ const NewHathorAlert = forwardRef(({ type, text, showAlert, fixedPosition }, ref
    * @param {number} duration - The display duration for the alert in milliseconds.
    */
   const show = duration => {
-    if (alertDiv.current) {
-      alertDiv.current.classList.add('show');
-      setTimeout(() => {
-        alertDiv.current.classList.remove('show');
-      }, duration);
+    // If the component is not in a fixed position, only handle the alert component
+    if (!fixedPosition) {
+      showAlertComponent();
+      return;
     }
-  };
 
-  useEffect(() => {
-    if (showAlert === undefined) {
+    // No-op if the container is unavailable
+    if (!containerDiv.current) {
       return;
     }
-    if (showAlert) {
+
+    // The component is in a fixed position: managing its parent container to allow for animations
+    // and precise positioning
+    containerDiv.current.classList.add('show');
+    setTimeout(showAlertComponent, 100);
+    setTimeout(() => {
+      // By the time the timeout is called, the container may have been unmounted
+      if (containerDiv?.current) {
+        containerDiv.current.classList.remove('show');
+      }
+    }, duration + 500);
+
+    /**
+     * Handles the Alert component display and removal
+     */
+    function showAlertComponent() {
+      // No-op if the ref is unavailable
+      if (!alertDiv.current) {
+        return;
+      }
+
       alertDiv.current.classList.add('show');
-    } else {
-      alertDiv.current.classList.remove('show');
+      setTimeout(() => {
+        // By the time the timeout is called, the component may have been unmounted
+        if (alertDiv?.current) {
+          alertDiv.current.classList.remove('show');
+        }
+      }, duration);
     }
-  }, [showAlert]);
+  };
 
   useImperativeHandle(ref, () => ({
     show,
   }));
 
-  const fixedPositionClass = fixedPosition ? 'fixed-position' : '';
-  return (
-    <div
-      ref={alertDiv}
-      className={`new-hathor-alert alert alert-${type} alert-dismissible fade ${fixedPositionClass}`}
-      role="alert"
-      style={{ display: 'flex', flexDirection: 'row' }}
-    >
-      <div className="success-icon">{type === 'success' ? <SuccessIcon /> : null}</div>
-      <p className="success-txt">{text}</p>
-    </div>
-  );
+  /**
+   * Renders the alert component
+   * @returns {Element}
+   */
+  function renderAlertDiv() {
+    return (
+      <div
+        ref={alertDiv}
+        className={`new-hathor-alert alert alert-${type} alert-dismissible fade`}
+        role="alert"
+        style={{ display: 'flex', flexDirection: 'row' }}
```

Comment:
I tried to avoid this `style` from the previous implementation, but the fade animations just wouldn't work without it. Decided to leave this cleanup to a future refactor.